### ISSUE TITLE
With makes where-clause parameters turn into `_`

### ISSUE THREAD

**Issue by JoeyEremondi** [idris-lang/Idris2#835]
This might be related to #834, but I'm not sure.

# Steps to Reproduce
```idris
import Data.Nat

foo : {P : Nat -> Type}  -> Dec (P 0)
foo = ?rhs1
  where
      bar : {P2 : Nat -> Type}
        -> (x : Nat)
        -> P2 x
      bar x with (S x)
      bar x | x1 = ?rhs2
```

# Expected Behavior
The parameter types of the where-bound function should show up like normal
```
Main> :t rhs2
   P : Nat -> Type
   P2 : Nat -> Type
   x1 : Nat
   x : Nat
------------------------------
rhs2 : P2 x
```

# Observed Behavior
`P2` in the goal is printed as `_`.
Also, P2 is missing in the parameter list (same as #834)
```
Main> :t rhs2
   P : Nat -> Type
   x1 : Nat
   x : Nat
------------------------------
rhs2 : _ x
```

**Comment by gallais** [comment]
> Also, P2 is missing in the parameter list

Based on logging, it seems that `P2` is somehow replaced with a machine-generated
`_` name. This does not happen if e.g. `bar` is made top-level.
I am trying to understand where this happens.

**Comment by gallais** [comment]
Fixed by #1017
